pkg/cmd/pipelineresource/create.go

```diff
@@ -0,0 +1,551 @@
+				caSecret.FieldName = "cadata"
+				res.pipelineResource.Spec.SecretParams = append(res.pipelineResource.Spec.SecretParams, caSecret)
+			}
+		}
+	}
+	return nil
+}
+
+func (res *resource) askPullRequestParams() error {
+	urlParam, err := askParam("url", res.askOpts)
+	if err != nil {
+		return err
+	}
+	if urlParam.Name != "" {
+		res.pipelineResource.Spec.Params = append(res.pipelineResource.Spec.Params, urlParam)
+	}
+
+	//ask for the secrets
+	qsOpts := []string{"Yes", "No"}
+	qs := "Do you want to set secrets ?"
+
+	ans, e := askToSelect(qs, qsOpts, res.askOpts)
+	if e != nil {
+		return e
+	}
+	if ans == qsOpts[1] {
+		return nil
+	}
+
+	secret, err := askSecret("githubToken", res.askOpts)
+	if err != nil {
+		return err
+	}
+	res.pipelineResource.Spec.SecretParams = append(res.pipelineResource.Spec.SecretParams, secret)
+
+	return nil
+}
+
+func (res *resource) askCloudEventParams() error {
+	targetURIParam, err := askParam("targetURI", res.askOpts)
+	if err != nil {
+		return err
+	}
+	if targetURIParam.Name != "" {
+		res.pipelineResource.Spec.Params = append(res.pipelineResource.Spec.Params, targetURIParam)
+	}
+	return nil
+}
+
+func askParam(paramName string, askOpts survey.AskOpt) (v1alpha1.ResourceParam, error) {
+	var param v1alpha1.ResourceParam
+	var qs = []*survey.Question{{
+		Name: "value",
+		Prompt: &survey.Input{
+			Message: fmt.Sprintf("Enter a value for %s : ", paramName),
+		},
+	}}
+
+	err := survey.Ask(qs, &param, askOpts)
+	if err != nil {
+		return param, Error(err)
+	}
+
+	if param.Value != "" {
+		param.Name = paramName
+	}
+
+	return param, nil
+}
+
+func askSecret(secret string, askOpts survey.AskOpt) (v1alpha1.SecretParam, error) {
+	var secrect v1alpha1.SecretParam
+	secrect.FieldName = secret
+	var qs = []*survey.Question{
+		{
+			Name: "secretKey",
+			Prompt: &survey.Input{
+				Message: fmt.Sprintf("Secret Key for %s :", secret),
+			},
+		},
+		{
+			Name: "secretName",
+			Prompt: &survey.Input{
+				Message: fmt.Sprintf("Secret Name for %s :", secret),
+			},
+		},
+	}
+
+	err := survey.Ask(qs, &secrect, askOpts)
+	if err != nil {
+		return secrect, Error(err)
+	}
+
+	return secrect, nil
+}
+
+func askToSelect(message string, options []string, askOpts survey.AskOpt) (string, error) {
+	var ans string
+	var qs1 = []*survey.Question{{
+		Name: "params",
+		Prompt: &survey.Select{
+			Message: message,
+			Options: options,
+		},
+	}}
+
+	err := survey.Ask(qs1, &ans, askOpts)
+	if err != nil {
+		return "", Error(err)
+	}
+
+	return ans, nil
+}
+
+func allResourceType() []string {
+	var resType []string
+
+	for _, val := range v1alpha1.AllResourceTypes {
+		resType = append(resType, string(val))
+	}
+
+	sort.Strings(resType)
+	return resType
+}
+
+func cast(answer string) v1alpha1.PipelineResourceType {
+	return v1alpha1.PipelineResourceType(answer)
+}
+
+func Error(err error) error {
+	switch err.Error() {
+	case "interrupt":
+		fmt.Println("interrupt")
+		os.Exit(1)
```

Comment:
This scares me a bit… On `docker/cli` we are using errors to do the `os.Exit` *only* on the `cmd/` package.
See https://github.com/docker/cli/blob/master/cmd/docker/docker.go#L292:L305 (could be done better but still :stuck_out_tongue_closed_eyes: )

Reply:
hm... then I will simply print a message and return an error, will update the PR.

Reply:
Now it returns `Error: interrupt`
@vdemeester WDYT